drawer-view/DrawerView.swift

```diff
@@ -402,3 +407,19 @@ private extension DrawerView {
         sholdRecalculateConstraints = false
     }
 }
+
+extension DrawerView : UIGestureRecognizerDelegate {
+    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldReceive touch: UITouch) -> Bool {
+        for subView in self.subviews {
+            guard let subView = subView as? DrawerViewTouchHandling else { continue }
+            if subView.interceptsTouch(touch) {
+                return false
+            }
+        }
+        return true
+    }
+}
+
+@objc public protocol DrawerViewTouchHandling {
```

Comment:
I will slightly change the name for the delegate method, in order to follow more traditional style for delegate protocols in Apple Platforms. The change will be made in a way that the method will look like this:

```swift
@objc func drawerViewInterceptsTouch(_ touch: UITouch) -> Bool
``` 

There are couple of reasons to do that:

1. Naming collisions, which rarely happen, but it's better to be safe than sorry 😄
2. Declaration that this method has direct relationship to the `drawer-view` dependency